ProjectFiles/Assets/Tools/Roadster/Editor/BezierSplineInspector copy.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+
+[CustomEditor(typeof(BezierSpline))] 
+public class BezierSplineInspector : Editor
+{
+    private BezierSpline spline;
+    private Transform handleTransform;
+    private Quaternion handleRotation;
+    
+    private const int lineSteps = 10;
+    
+    private const float handleSize = 0.04f;
+    private const float pickSize = 0.06f;
+	
+    private int selectedIndex = -1;
+
+    
+    private void OnSceneGUI () {
+        //Get the bezier component
+        spline = target as BezierSpline;
+
+        //Get orientation of handle
+        handleTransform = spline.transform;
+        handleRotation = Tools.pivotRotation == PivotRotation.Local ? handleTransform.rotation : Quaternion.identity;
+        
+        //Display the points
+        Vector3 p0 = ShowPoint(0);
+        for (int i = 1; i < spline.ControlPointCount; i += 3) {
+            Vector3 p1 = ShowPoint(i);
+            Vector3 p2 = ShowPoint(i+1);
+            Vector3 p3 = ShowPoint(i+2);
+            
+            //Draw lines between points
+            Handles.color = Color.gray;
+            Handles.DrawLine(p0, p1);
+            Handles.DrawLine(p1, p2);
+            Handles.DrawLine(p2, p3);
+        
+            //Draw aproximate curve
+            Handles.color = Color.white;
+            
+            //Draw interpolated bezier curve
+//            int lineSteps = 10;
+//            Vector3 lineStart = curve.GetPoint(0f);
+//            for (int i = 1; i <= lineSteps; i++) {
+//                Vector3 lineEnd = curve.GetPoint(i / (float)lineSteps);
+//                Handles.DrawLine(lineStart, lineEnd);
+//                lineStart = lineEnd;
+//            }
+            Handles.DrawBezier(p0, p3, p1, p2, Color.white, null, 2f);
+            p0 = p3;
+        }
+    }    
+    
+    private static Color[] modeColors = {
```

Comment:
top of the file?

Reply:
Moving it, not sure why I put it there...